pkg/kotsadm/namespaces.go

```diff
@@ -0,0 +1,66 @@
+package kotsadm
+
+import (
+	"github.com/pkg/errors"
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	"github.com/replicatedhq/kots/kotskinds/client/kotsclientset/scheme"
+	"github.com/replicatedhq/kots/pkg/kotsadm/types"
+	"github.com/replicatedhq/kots/pkg/logger"
+	corev1 "k8s.io/api/core/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func ensureAdditionalNamespaces(deployOptions *types.DeployOptions, clientset *kubernetes.Clientset, log *logger.Logger) error {
+	// try to parse
+	if deployOptions.ApplicationMetadata == nil {
+		return nil
+	}
+
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	obj, gvk, err := decode(deployOptions.ApplicationMetadata, nil, nil)
+	if err != nil {
+		return nil // no error here
+	}
+
+	if gvk.Group != "kots.io" || gvk.Version != "v1beta1" || gvk.Kind != "Application" {
+		return nil
+	}
+
+	application := obj.(*kotsv1beta1.Application)
+
+	if application.Spec.AdditionalNamespaces == nil {
+		return nil
+	}
+
+	for _, additionalNamespace := range application.Spec.AdditionalNamespaces {
+		// We support "*" for additional namespaces to handle pullsecret propagation
```

Comment:
"We `don't` support", right?

Reply:
We do, but we don't create that.  It (will be) handled in the kotsadm code to handle imagePulLSecret propagation

Reply:
Does this then break on something like this:
```
 - custom_ns
 - worker*
```

Reply:
unsupported.  only "*" and/or specific (exact) namespaces